### ISSUE TITLE
Aggregation metrics ignore zeros tensors

### ISSUE THREAD

**Issue by mszulc913** [PyTorchLightning/metrics#1063]
## 🐛 Bug

When feeding `CatMetric`, `MinMetric` or `MaxMetric` with zeros tensor, the metric isn't updated.

### To Reproduce

1. Create a new `CatMetric` instance and feed it with zeros tensor.
2. `compute()` returns empty list.

#### Code sample

```python
import torch
import torchmetrics

metric = torchmetrics.CatMetric()
metric(torch.zeros((5, )))
print(metric.compute()) # outputs []
```

### Expected behavior

Zeros tensor should update metric's state.

### Environment

```
torch==1.11.0
torchmetrics==0.9.0
```
Ubuntu 20.04.4

### Additional context
The problem is caused by these lines:
https://github.com/PyTorchLightning/metrics/blob/567fa3405d5a60ba0a6052e57afe5a5d34c376a7/torchmetrics/aggregation.py#L139
https://github.com/PyTorchLightning/metrics/blob/567fa3405d5a60ba0a6052e57afe5a5d34c376a7/torchmetrics/aggregation.py#L188
https://github.com/PyTorchLightning/metrics/blob/567fa3405d5a60ba0a6052e57afe5a5d34c376a7/torchmetrics/aggregation.py#L280

`any(value.flatten())` evaluates to `False` on zeros tensors.
